algorithms/poc.js

```diff
@@ -104,49 +59,52 @@ const createTick = (lastOHLCV, prevOpen, prevClose) => {
     return tick;
 };
 
-// Initialize coinData
-const initialize = async (base, account, coinData, binanceClient, symbol) => {
-    // request ticker info & initialize previous price with live price
-    const symbolOHLV = await binanceClient.fetchOHLCV(symbol);
-    const lastOHLCV = symbolOHLV[symbolOHLV.length - 1];
+// Interval function
+const tick = async (coinData, account, binanceClient, symbol, base) => {
+    /** request ticker info & initialize previous price with live price
+     *  Note that the info from the last (current) candle may be incomplete until the candle is closed (until the next candle starts).
+     *
+     *  [
+     *      1504541580000, // UTC timestamp in milliseconds, integer
+     *      4235.4,        // (O)pen price, float
+     *      4240.6,        // (H)ighest price, float
+     *      4230.0,        // (L)owest price, float
+     *      4230.7,        // (C)losing price, float
+     *      37.72941911    // (V)olume (in terms of the base currency), float
+     *  ]
+     */
+    const symbolOHLCV = await binanceClient.fetchOHLCV(symbol);
+    const initialOHLCV = symbolOHLCV[0];
 
     let initializeTick = createTick(
-        lastOHLCV,
-        lastOHLCV[1], // open
-        lastOHLCV[4] // close
+        initialOHLCV,
+        initialOHLCV[1], // open
+        initialOHLCV[4] // close
     );
     coinData.previous = initializeTick;
-    console.log(coinData);
 
     // request account balance & initialize account information
     const accountBalance = await binanceClient.fetchBalance();
     account.startingBalance = accountBalance.free[base];
     account.theoryBalance = account.startingBalance;
     account.binanceFee = 1 - accountBalance.info.takerCommission / 10000;
-};
-
-// Interval function
-const tick = async (coinData, account, dataFilePath, binanceClient, symbol) => {
-    const { open, close } = coinData.previous.heikinAshi;
-
-    // request datetime from ticker from API
-    const symbolTicker = await binanceClient.fetchTicker(symbol);
-    const ping = { time: symbolTicker.datetime, account, coinData };
 
-    // use this API for the OHLC prices
-    const symbolOHLV = await binanceClient.fetchOHLCV(symbol);
-    const lastOHLCV = symbolOHLV[symbolOHLV.length - 1];
-    let intervalTick = createTick(lastOHLCV, open, close);
-    coinData.current = intervalTick;
+    // this could work, need to map and take an average of every 5 instances
+    symbolOHLCV.forEach((lastOHLCV, index) => {
```

Comment:
This needs to be reconfigured to collect  X amount of candles and factor based on that average:
O: last
H: max
L: min
C: first